src/main/java/com/jcabi/s3/Bucket.java

```diff
@@ -183,9 +183,16 @@ public void remove(final String key) throws IOException {
                     new Function<String, String>() {
                         @Override
                         public String apply(final String input) {
-                            return input.substring(
-                                Bucket.Prefixed.this.prefix.length()
-                            );
+                            final String result;
+                            final String head = Bucket.Prefixed.this.prefix;
+                            if ((input.length() < head.length())
+                                && (head.contains(input))
+                            ) {
+                                result = "/";
+                            } else {
+                                result = input.substring(head.length());
+                            }
+                            return result;
```

Comment:
@yegor256 that is because I found that if `Bucket.Prefixed` used in the test is not capable to show up root directory with prefix if `MKBucket` return so. Should I basically get rid of `Bucket.Prefixed` usage in the test and create a new ticket on `Bucket.Prefixed` or keeping this change? Thanks.